apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -496,6 +496,9 @@ export default class InstallManager {
       }
     }
 
+    // @todo for PNPM, we should look at the lock file to determine if the store needs to be cleaned
+    // ie even if the marker file does not exist, the store may be ok if there is no lock file
```

Comment:
I would err on the side of simple+reliable for now, and add complicated edge case optimizations later if we actually see a measurable need